example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java

```diff
@@ -142,6 +154,86 @@ public void shouldRegisterRecipeAddedEvent() {
     }
 
     @Test
+    public final void singleAddAndRemoveRecipe() throws InterruptedException {
+        shouldRegisterRecipeRemovedEvent();
+    }
+
+
+    @Test
+    public final void sequentialAddAndRemoveRecipe() throws InterruptedException {
+        shouldRegisterRecipeRemovedEvent();
+        shouldRegisterRecipeRemovedEvent();
+        shouldRegisterRecipeRemovedEvent();
+        shouldRegisterRecipeRemovedEvent();
+        shouldRegisterRecipeRemovedEvent();
+    }
+
+    @Test
+    public final void concurrentAddAndRemoveRecipe() throws InterruptedException {
+        ExecutorService exec = Executors.newFixedThreadPool(5);
+        for (int i = 0; i <10; i++) {
+            exec.execute(new Runnable() {
+                @Override
+                public void run() {
+                    shouldRegisterRecipeRemovedEvent();
+                }
+            });
+        }
+        exec.shutdown();
+        exec.awaitTermination(180, TimeUnit.SECONDS);
+    }
+
+
+    public void shouldRegisterRecipeRemovedEvent() {
+        LOGGER.debug("Starting with ADD");
+        final String  recipeId = UUID.randomUUID().toString();
+
+        sendTo(RECIPES_RESOURCE_URI + recipeId).request()
+                .post(entity(
+                        jsonObject()
+                                .add("name", "Vanilla cake")
+                                .add("glutenFree", false)
+                                .add("ingredients", createArrayBuilder()
+                                        .add(createObjectBuilder()
+                                                .add("name", "vanilla")
+                                                .add("quantity", 2)
+                                        ).build()
+                                ).build().toString(),
+                        ADD_RECIPE_MEDIA_TYPE));
+
+        await().until(() -> eventsWithPayloadContaining(recipeId).count() == 1);
+
+        EventLog event = eventsWithPayloadContaining(recipeId).findFirst().get();
+        assertThat(event.getName(), is("cakeshop.recipe-added"));
+        with(event.getMetadata())
+                .assertEquals("stream.id", recipeId)
+                .assertEquals("stream.version", 1);
+        String eventPayload = event.getPayload();
+        with(eventPayload)
+                .assertThat("$.recipeId", equalTo(recipeId))
+                .assertThat("$.name", equalTo("Vanilla cake"))
+                .assertThat("$.glutenFree", equalTo(false))
+                .assertThat("$.ingredients[0].name", equalTo("vanilla"))
+                .assertThat("$.ingredients[0].quantity", equalTo(2));
+        LOGGER.debug("Ending with ADD");
```

Comment:
Not sure this debugging comes out anywhere?

Consider if we need it or not (and the logger at the top if we don't)